packages/nimbus/src/components/date-range-picker/date-range-picker.dev.mdx

```diff
@@ -0,0 +1,235 @@
+---
+title: DateRangePicker Component
+tab-title: Implementation
+tab-order: 3
+---
+
+# DateRangePicker Component
+
+The DateRangePicker component combines DateInput fields with a RangeCalendar popover for date range selection. Users can either type date ranges directly or select from the calendar interface, providing flexible input methods for date range scenarios.
+
+## Component Anatomy
+
+The DateRangePicker component is built with several integrated parts:
+
+```
+DateRangePicker
+├── DateRangePickerRootSlot (wrapper)
+├── DateRangePickerGroupSlot (input container)
+│   ├── DateInput (start)
+│   ├── Text (separator "–")
+│   ├── DateInput (end)
+│   └── DateRangePickerTriggerSlot
+│       ├── IconButton (clear)
+│       └── IconButton (calendar toggle)
+└── DateRangePickerPopoverSlot
+    ├── DateRangePickerCalendarSlot
+    │   └── RangeCalendar
+    └── DateRangePickerTimeInput (when granularity includes time)
```

Comment:
Is a file tree what we want here? or a png calling out each of the sub-components visually? ([like react-aria has](https://react-spectrum.adobe.com/react-aria/DateRangePicker.html#anatomy))